cli/command/image/push.go

```diff
@@ -70,6 +72,11 @@ func NewPushCommand(dockerCli command.Cli) *cobra.Command {
 //
 //nolint:gocyclo
 func RunPush(ctx context.Context, dockerCli command.Cli, opts pushOptions) error {
+	// Don't default to DOCKER_DEFAULT_PLATFORM env variable on unsupported API versions.
+	if versions.LessThan(dockerCli.CurrentVersion(), "1.46") {
```

Comment:
The only thing I'm thinking here is if we should unconditionally set it to empty, or only if `!flags.Chanced("platform")`  (i.e., user didn't explicitly set a `--platform` flag), because we also have some code in the client to print an error when trying to use the option on older API versions; https://github.com/docker/cli/blob/9bb1a62735174e9220d84fecc056a0ef8a1fc26f/vendor/github.com/docker/docker/client/image_push.go#L41-L44

Reply:
Actually even wondering if this platform should default to `DOCKER_DEFAULT_PLATFORM` or if it just shouldn't pick up that value at all; I need to find back the reason that env-var was added, but it may have been for LCOW so that users could set a default platform for `docker run` (and thus `docker pull`). Maybe that doesn't make sense for `docker push`, because that should probably default to pushing what's there.

Reply:
> because we also have some code in the client to print an error when trying to use the option on older API versions;

Yep this is exactly why I didn't handle `Changed` here - passing `--platform` on older API will error out earlier anyway.

> Actually even wondering if this platform should default to `DOCKER_DEFAULT_PLATFORM` or if it just shouldn't pick up that value at all;

Yeah, considering that it's also a newly introduced flag, I would also be fine with just ignoring that env variable.

Reply:
> Yeah, considering that it's also a newly introduced flag, I would also be fine with just ignoring that env variable.

Yup, was thinking that as well; that `addPlatformFlag` utility is a bit awkward in either case, so just adding a flag "as usual" that doesn't have the magic `os.GetEnv()` default seems fine with me.

Reply:
Done